### ISSUE TITLE
2.5.0: pytest is failing

### ISSUE THREAD

**Issue by kloczek** [pytest-dev/pytest-xdist#742]
I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w`
- install .whl file in </install/prefix>
- "pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- during testing I'm disabling few pytest extensions which in other tickets have been identified as causing some issues

Here is pytest output:
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-xdist-2.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-pytest-xdist-2.5.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -p no:randomly -p no:benchmark -p no:django -p no:twisted --deselect testing/acceptance_test.py::test_issue_594_random_parametrize --deselect testing/test_newhooks.py::TestHooks::test_node_collection_finished
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-0.13.1
rootdir: /home/tkloczko/rpmbuild/BUILD/pytest-xdist-2.5.0, configfile: tox.ini, testpaths: testing
plugins: xdist-2.5.0, mock-3.6.1, cov-2.12.1, anyio-3.3.4, flaky-3.7.0, console-scripts-1.2.0, asyncio-0.16.0, freezegun-0.4.2, flake8-1.0.7, rerunfailures-9.1.1, yagot-0.5.0, forked-1.4.0, ordering-0.6, Faker-10.0.0, pyfakefs-4.5.3, datadir-1.3.1, regressions-2.2.0, timeout-2.0.2, perf-0.10.1, trio-0.7.0, requests-mock-1.9.3, hypothesis-6.31.5, easy-server-0.8.0
collected 167 items / 2 deselected / 165 selected

testing/acceptance_test.py ..........F......x.......xx.......................x.........................................                                              [ 55%]
testing/test_dsession.py ........x...x                                                                                                                               [ 63%]
testing/test_looponfail.py ...........x...                                                                                                                           [ 72%]
testing/test_newhooks.py F.                                                                                                                                          [ 73%]
testing/test_plugin.py ..............                                                                                                                                [ 82%]
testing/test_remote.py x...Fx......                                                                                                                                  [ 89%]
testing/test_workermanage.py ........x.......s                                                                                                                       [100%]

================================================================================= FAILURES =================================================================================
_______________________________________________________________ TestDistribution.test_dist_tests_with_crash ________________________________________________________________

self = <acceptance_test.TestDistribution object at 0x7fcf8f6e0070>, pytester = <Pytester PosixPath('/tmp/pytest-of-tkloczko/pytest-10/test_dist_tests_with_crash0')>

    @pytest.mark.xfail("sys.platform.startswith('java')", run=False)
    def test_dist_tests_with_crash(self, pytester: pytest.Pytester) -> None:
        if not hasattr(os, "kill"):
            pytest.skip("no os.kill")

        p1 = pytester.makepyfile(
            """
                import pytest
                def test_fail0():
                    assert 0
                def test_fail1():
                    raise ValueError()
                def test_ok():
                    pass
                def test_skip():
                    pytest.skip("hello")
                def test_crash():
                    import time
                    import os
                    time.sleep(0.5)
                    os.kill(os.getpid(), 15)
            """
        )
        result = pytester.runpytest(p1, "-v", "-d", "-n1")
>       result.stdout.fnmatch_lines(
            [
                "*Python*",
                "*PASS**test_ok*",
                "*node*down*",
                "*3 failed, 1 passed, 1 skipped*",
            ]
        )
E       Failed: nomatch: '*Python*'
E           and: '============================= test session starts =============================='
E       fnmatch: '*Python*'
E          with: 'platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-0.13.1 -- /usr/bin/python3'
E       nomatch: '*PASS**test_ok*'
E           and: 'cachedir: .pytest_cache'
E           and: 'benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)'
E           and: 'Using --randomly-seed=244357277'
E           and: "hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/tkloczko/rpmbuild/BUILD/pytest-xdist-2.5.0/.hypothesis/examples')"
E           and: 'rootdir: /tmp/pytest-of-tkloczko/pytest-10/test_dist_tests_with_crash0'
E           and: 'plugins: xdist-2.5.0, mock-3.6.1, cov-2.12.1, anyio-3.3.4, flaky-3.7.0, console-scripts-1.2.0, asyncio-0.16.0, freezegun-0.4.2, flake8-1.0.7, rerunfailures-9.1.1, yagot-0.5.0, forked-1.4.0, ordering-0.6, Faker-10.0.0, benchmark-3.4.1, pyfakefs-4.5.3, datadir-1.3.1, regressions-2.2.0, timeout-2.0.2, randomly-3.10.3, perf-0.10.1, trio-0.7.0, requests-mock-1.9.3, hypothesis-6.31.5, easy-server-0.8.0'
E           and: 'gw0 I'
E           and: ''
E           and: '[gw0] linux Python 3.8.12 cwd: /tmp/pytest-of-tkloczko/pytest-10/test_dist_tests_with_crash0'
E           and: ''
E           and: '[gw0] Python 3.8.12 (default, Dec 17 2021, 08:35:49)  -- [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)]'
E           and: 'gw0 [5]'
E           and: ''
E           and: 'scheduling tests via LoadScheduling'
E           and: ''
E           and: 'test_dist_tests_with_crash.py::test_fail0 '
E           and: '[gw0] [ 20%] FAILED test_dist_tests_with_crash.py::test_fail0 '
E           and: 'test_dist_tests_with_crash.py::test_fail1 '
E           and: '[gw0] [ 40%] FAILED test_dist_tests_with_crash.py::test_fail1 '
E           and: 'test_dist_tests_with_crash.py::test_crash '
E           and: '[gw0] node down: Not properly terminated'
E           and: '[gw0] [ 60%] FAILED test_dist_tests_with_crash.py::test_crash '
E           and: ''
E           and: 'replacing crashed worker gw0'
E           and: ''
E           and: '[gw1] linux Python 3.8.12 cwd: /tmp/pytest-of-tkloczko/pytest-10/test_dist_tests_with_crash0'
E           and: ''
E           and: '[gw1] Python 3.8.12 (default, Dec 17 2021, 08:35:49)  -- [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)]'
E           and: ''
E           and: 'test_dist_tests_with_crash.py::test_skip '
E           and: '[gw1] [ 80%] SKIPPED test_dist_tests_with_crash.py::test_skip '
E           and: 'test_dist_tests_with_crash.py::test_ok '
E       fnmatch: '*PASS**test_ok*'
E          with: '[gw1] [100%] PASSED test_dist_tests_with_crash.py::test_ok '
E       nomatch: '*node*down*'
E           and: ''
E           and: '=================================== FAILURES ==================================='
E           and: '__________________________________ test_fail0 __________________________________'
E           and: '[gw0] linux -- Python 3.8.12 /usr/bin/python3'
E           and: ''
E           and: '    def test_fail0():'
E           and: '>       assert 0'
E           and: 'E       assert 0'
E           and: ''
E           and: 'test_dist_tests_with_crash.py:3: AssertionError'
E           and: '__________________________________ test_fail1 __________________________________'
E           and: '[gw0] linux -- Python 3.8.12 /usr/bin/python3'
E           and: ''
E           and: '    def test_fail1():'
E           and: '>       raise ValueError()'
E           and: 'E       ValueError'
E           and: ''
E           and: 'test_dist_tests_with_crash.py:5: ValueError'
E           and: '________________________ test_dist_tests_with_crash.py _________________________'
E           and: '[gw0] linux -- Python 3.8.12 /usr/bin/python3'
E           and: "worker 'gw0' crashed while running 'test_dist_tests_with_crash.py::test_crash'"
E           and: '=========================== short test summary info ============================'
E           and: 'FAILED test_dist_tests_with_crash.py::test_fail0 - assert 0'
E           and: 'FAILED test_dist_tests_with_crash.py::test_fail1 - ValueError'
E           and: 'FAILED test_dist_tests_with_crash.py::test_crash'
E           and: '==================== 3 failed, 1 passed, 1 skipped in 6.52s ===================='
E       remains unmatched: '*node*down*'

/home/tkloczko/rpmbuild/BUILD/pytest-xdist-2.5.0/testing/acceptance_test.py:181: Failed
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
============================= test session starts ==============================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=244357277
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/tkloczko/rpmbuild/BUILD/pytest-xdist-2.5.0/.hypothesis/examples')
rootdir: /tmp/pytest-of-tkloczko/pytest-10/test_dist_tests_with_crash0
plugins: xdist-2.5.0, mock-3.6.1, cov-2.12.1, anyio-3.3.4, flaky-3.7.0, console-scripts-1.2.0, asyncio-0.16.0, freezegun-0.4.2, flake8-1.0.7, rerunfailures-9.1.1, yagot-0.5.0, forked-1.4.0, ordering-0.6, Faker-10.0.0, benchmark-3.4.1, pyfakefs-4.5.3, datadir-1.3.1, regressions-2.2.0, timeout-2.0.2, randomly-3.10.3, perf-0.10.1, trio-0.7.0, requests-mock-1.9.3, hypothesis-6.31.5, easy-server-0.8.0
gw0 I
[gw0] linux Python 3.8.12 cwd: /tmp/pytest-of-tkloczko/pytest-10/test_dist_tests_with_crash0
[gw0] Python 3.8.12 (default, Dec 17 2021, 08:35:49)  -- [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)]
gw0 [5]

scheduling tests via LoadScheduling

test_dist_tests_with_crash.py::test_fail0
[gw0] [ 20%] FAILED test_dist_tests_with_crash.py::test_fail0
test_dist_tests_with_crash.py::test_fail1
[gw0] [ 40%] FAILED test_dist_tests_with_crash.py::test_fail1
test_dist_tests_with_crash.py::test_crash
[gw0] node down: Not properly terminated
[gw0] [ 60%] FAILED test_dist_tests_with_crash.py::test_crash

replacing crashed worker gw0
[gw1] linux Python 3.8.12 cwd: /tmp/pytest-of-tkloczko/pytest-10/test_dist_tests_with_crash0
[gw1] Python 3.8.12 (default, Dec 17 2021, 08:35:49)  -- [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)]

test_dist_tests_with_crash.py::test_skip
[gw1] [ 80%] SKIPPED test_dist_tests_with_crash.py::test_skip
test_dist_tests_with_crash.py::test_ok
[gw1] [100%] PASSED test_dist_tests_with_crash.py::test_ok

=================================== FAILURES ===================================
__________________________________ test_fail0 __________________________________
[gw0] linux -- Python 3.8.12 /usr/bin/python3

    def test_fail0():
>       assert 0
E       assert 0

test_dist_tests_with_crash.py:3: AssertionError
__________________________________ test_fail1 __________________________________
[gw0] linux -- Python 3.8.12 /usr/bin/python3

    def test_fail1():
>       raise ValueError()
E       ValueError

test_dist_tests_with_crash.py:5: ValueError
________________________ test_dist_tests_with_crash.py _________________________
[gw0] linux -- Python 3.8.12 /usr/bin/python3
worker 'gw0' crashed while running 'test_dist_tests_with_crash.py::test_crash'
=========================== short test summary info ============================
FAILED test_dist_tests_with_crash.py::test_fail0 - assert 0
FAILED test_dist_tests_with_crash.py::test_fail1 - ValueError
FAILED test_dist_tests_with_crash.py::test_crash
==================== 3 failed, 1 passed, 1 skipped in 6.52s ====================
_____________________________________________________________________ TestHooks.test_runtest_logreport _____________________________________________________________________

self = <test_newhooks.TestHooks object at 0x7fcf4875c0d0>, pytester = <Pytester PosixPath('/tmp/pytest-of-tkloczko/pytest-10/test_runtest_logreport0')>

    def test_runtest_logreport(self, pytester: pytest.Pytester) -> None:
        """Test that log reports from pytest_runtest_logreport when running
        with xdist contain "node", "nodeid", "worker_id", and "testrun_uid" attributes. (#8)
        """
        pytester.makeconftest(
            """
            def pytest_runtest_logreport(report):
                if hasattr(report, 'node'):
                    if report.when == "call":
                        workerid = report.node.workerinput['workerid']
                        testrunuid = report.node.workerinput['testrunuid']
                        if workerid != report.worker_id:
                            print("HOOK: Worker id mismatch: %s %s"
                                   % (workerid, report.worker_id))
                        elif testrunuid != report.testrun_uid:
                            print("HOOK: Testrun uid mismatch: %s %s"
                                   % (testrunuid, report.testrun_uid))
                        else:
                            print("HOOK: %s %s %s"
                                   % (report.nodeid, report.worker_id, report.testrun_uid))
        """
        )
        res = pytester.runpytest("-n1", "-s")
>       res.stdout.fnmatch_lines(
            [
                "*HOOK: test_runtest_logreport.py::test_a gw0 *",
                "*HOOK: test_runtest_logreport.py::test_b gw0 *",
                "*HOOK: test_runtest_logreport.py::test_c gw0 *",
                "*3 passed*",
            ]
        )
E       Failed: nomatch: '*HOOK: test_runtest_logreport.py::test_a gw0 *'
E           and: '============================= test session starts =============================='
E           and: 'platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-0.13.1'
E           and: 'benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)'
E           and: 'Using --randomly-seed=2699028449'
E           and: 'rootdir: /tmp/pytest-of-tkloczko/pytest-10/test_runtest_logreport0'
E           and: 'plugins: xdist-2.5.0, mock-3.6.1, cov-2.12.1, anyio-3.3.4, flaky-3.7.0, console-scripts-1.2.0, asyncio-0.16.0, freezegun-0.4.2, flake8-1.0.7, rerunfailures-9.1.1, yagot-0.5.0, forked-1.4.0, ordering-0.6, Faker-10.0.0, benchmark-3.4.1, pyfakefs-4.5.3, datadir-1.3.1, regressions-2.2.0, timeout-2.0.2, randomly-3.10.3, perf-0.10.1, trio-0.7.0, requests-mock-1.9.3, hypothesis-6.31.5, easy-server-0.8.0'
E           and: 'gw0 I'
E           and: 'gw0 [3]'
E           and: ''
E           and: '.HOOK: test_runtest_logreport.py::test_b gw0 0d69e931b31a4f4a8e735485eb0afc7d'
E           and: '.HOOK: test_runtest_logreport.py::test_c gw0 0d69e931b31a4f4a8e735485eb0afc7d'
E       fnmatch: '*HOOK: test_runtest_logreport.py::test_a gw0 *'
E          with: '.HOOK: test_runtest_logreport.py::test_a gw0 0d69e931b31a4f4a8e735485eb0afc7d'
E       nomatch: '*HOOK: test_runtest_logreport.py::test_b gw0 *'
E           and: ''
E           and: '============================== 3 passed in 3.10s ==============================='
E       remains unmatched: '*HOOK: test_runtest_logreport.py::test_b gw0 *'

/home/tkloczko/rpmbuild/BUILD/pytest-xdist-2.5.0/testing/test_newhooks.py:39: Failed
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
============================= test session starts ==============================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-0.13.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=2699028449
rootdir: /tmp/pytest-of-tkloczko/pytest-10/test_runtest_logreport0
plugins: xdist-2.5.0, mock-3.6.1, cov-2.12.1, anyio-3.3.4, flaky-3.7.0, console-scripts-1.2.0, asyncio-0.16.0, freezegun-0.4.2, flake8-1.0.7, rerunfailures-9.1.1, yagot-0.5.0, forked-1.4.0, ordering-0.6, Faker-10.0.0, benchmark-3.4.1, pyfakefs-4.5.3, datadir-1.3.1, regressions-2.2.0, timeout-2.0.2, randomly-3.10.3, perf-0.10.1, trio-0.7.0, requests-mock-1.9.3, hypothesis-6.31.5, easy-server-0.8.0
gw0 I
gw0 [3]

.HOOK: test_runtest_logreport.py::test_b gw0 0d69e931b31a4f4a8e735485eb0afc7d
.HOOK: test_runtest_logreport.py::test_c gw0 0d69e931b31a4f4a8e735485eb0afc7d
.HOOK: test_runtest_logreport.py::test_a gw0 0d69e931b31a4f4a8e735485eb0afc7d

============================== 3 passed in 3.10s ===============================
__________________________________________________________________ TestWorkerInteractor.test_runtests_all __________________________________________________________________

self = <test_remote.TestWorkerInteractor object at 0x7fcf44835520>, worker = <test_remote.WorkerSetup object at 0x7fcf44877340>
unserialize_report = <function TestWorkerInteractor.unserialize_report.<locals>.unserialize at 0x7fcf43dadf70>

    def test_runtests_all(self, worker: WorkerSetup, unserialize_report) -> None:
        worker.pytester.makepyfile(
            """
            def test_func(): pass
            def test_func2(): pass
        """
        )
        worker.setup()
        ev = worker.popevent()
        assert ev.name == "workerready"
        ev = worker.popevent()
        assert ev.name == "collectionstart"
        assert not ev.kwargs
        ev = worker.popevent("collectionfinish")
        ids = ev.kwargs["ids"]
        assert len(ids) == 2
        worker.sendcommand("runtests_all")
        worker.sendcommand("shutdown")
        for func in "::test_func", "::test_func2":
            for i in range(3):  # setup/call/teardown
                ev = worker.popevent("testreport")
                assert ev.name == "testreport"
                rep = unserialize_report(ev.kwargs["data"])
>               assert rep.nodeid.endswith(func)
E               AssertionError: assert False
E                +  where False = <built-in method endswith of str object at 0x7fcf5125d390>('::test_func')
E                +    where <built-in method endswith of str object at 0x7fcf5125d390> = 'test_runtests_all.py::test_func2'.endswith
E                +      where 'test_runtests_all.py::test_func2' = <TestReport 'test_runtests_all.py::test_func2' when='setup' outcome='passed'>.nodeid

/home/tkloczko/rpmbuild/BUILD/pytest-xdist-2.5.0/testing/test_remote.py:182: AssertionError
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
skipping <EventCall logstart(**{'nodeid': 'test_runtests_all.py::test_func2', 'location': ('test_runtests_all.py', 1, 'test_func2')})>
============================================================================= warnings summary =============================================================================
testing/acceptance_test.py: 74 warnings
testing/test_dsession.py: 1 warning
testing/test_newhooks.py: 2 warnings
testing/test_remote.py: 4 warnings
  /usr/lib/python3.8/site-packages/pytest_benchmark/logger.py:46: PytestBenchmarkWarning: Benchmarks are automatically disabled because xdist plugin is active.Benchmarks cannot be performed reliably in a parallelized environment.
    warner(PytestBenchmarkWarning(text))

testing/acceptance_test.py: 80 warnings
testing/test_dsession.py: 10 warnings
testing/test_looponfail.py: 8 warnings
testing/test_newhooks.py: 2 warnings
testing/test_plugin.py: 16 warnings
testing/test_remote.py: 10 warnings
testing/test_workermanage.py: 13 warnings
  /usr/lib/python3.8/site-packages/pytest_randomly/__init__.py:50: UserWarning: The NumPy module was reloaded (imported a second time). This can in some cases result in small but subtle issues and is discouraged.
    from numpy import random as np_random

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================================= short test summary info ==========================================================================
SKIPPED [1] ../../../../../usr/lib/python3.8/site-packages/_pytest/config/__init__.py:1473: no 'gspecs' option found
XFAIL testing/acceptance_test.py::TestDistEach::test_simple_diffoutput
  reason: [NOTRUN] other python versions might not have pytest installed
XFAIL testing/acceptance_test.py::test_terminate_on_hangingnode
XFAIL testing/acceptance_test.py::test_session_hooks
  reason: [NOTRUN] works if run outside test suite
XFAIL testing/acceptance_test.py::TestNodeFailure::test_each_multiple
  #20: xdist race condition on node restart
XFAIL testing/test_dsession.py::TestDistReporter::test_rsync_printing
XFAIL testing/test_dsession.py::test_pytest_issue419
  duplicate test ids not supported yet
XFAIL testing/test_looponfail.py::TestLooponFailing::test_looponfail_removed_test
  broken by pytest 3.1+
XFAIL testing/test_remote.py::test_remoteinitconfig
  #59
XFAIL testing/test_remote.py::TestWorkerInteractor::test_happy_run_events_converted
  reason: implement a simple test for event production
XFAIL testing/test_workermanage.py::TestNodeManager::test_rsync_roots_no_roots
  reason: [NOTRUN]
FAILED testing/acceptance_test.py::TestDistribution::test_dist_tests_with_crash - Failed: nomatch: '*Python*'
FAILED testing/test_newhooks.py::TestHooks::test_runtest_logreport - Failed: nomatch: '*HOOK: test_runtest_logreport.py::test_a gw0 *'
FAILED testing/test_remote.py::TestWorkerInteractor::test_runtests_all - AssertionError: assert False
======================================= 3 failed, 151 passed, 1 skipped, 2 deselected, 10 xfailed, 220 warnings in 502.36s (0:08:22) =======================================
```

**Comment by kloczek** [comment]
ping .. 😋

**Comment by nicoddemus** [comment]
Hi @kloczek,

Took a look, seems like  `test_runtest_logreport` is failing due to ordering, it expected hooks to be called in this order:

```
"*HOOK: test_runtest_logreport.py::test_a gw0 *",
"*HOOK: test_runtest_logreport.py::test_b gw0 *",
"*HOOK: test_runtest_logreport.py::test_c gw0 *",
"*3 passed*",
```

However they were called in this order:

```
.HOOK: test_runtest_logreport.py::test_b gw0 0d69e931b31a4f4a8e735485eb0afc7d
.HOOK: test_runtest_logreport.py::test_c gw0 0d69e931b31a4f4a8e735485eb0afc7d
.HOOK: test_runtest_logreport.py::test_a gw0 0d69e931b31a4f4a8e735485eb0afc7d
```

So it seems the problem is the fact that you have `randomly` installed, as shown in the header:

```
Using --randomly-seed=2699028449
```

This is changing the order the internal tests are executed, breaking the outer test.

Didn't take a deep look at the other two, but given the large number of plugins installed in the same environment:

```
plugins: xdist-2.5.0, mock-3.6.1, cov-2.12.1, anyio-3.3.4, flaky-3.7.0, console-scripts-1.2.0, asyncio-0.16.0, freezegun-0.4.2, flake8-1.0.7, rerunfailures-9.1.1, yagot-0.5.0, forked-1.4.0, ordering-0.6, Faker-10.0.0, pyfakefs-4.5.3, datadir-1.3.1, regressions-2.2.0, timeout-2.0.2, perf-0.10.1, trio-0.7.0, requests-mock-1.9.3, hypothesis-6.31.5, easy-server-0.8.0
```

We test pytest-xdist without any other plugins in the environment, not because pytest-xdist does not work with other plugins when installed together in user's environments, but because we need a "sterile" test environment in order to be able to test corner cases, internal behaviors, and etc (the test above is an example of that).

We can't guarantee that the suite will run 100% with other plugins installed, as their combination is endless, and the effect on pytest-xdist's own tests might be unknown. To be clear: this does not mean pytest-xdist won't work with those plugins, it is just that pytest-xdist's *own test suite* might not work 100% because pytest-xdist's own suite tests internal behavior or edge cases.

I recommend to test pytest-xdist without any other plugins installed. 

One can look at the `tox.ini` file to figure out which plugins get installed in each test environment (this is true in general for most pytest plugins tested with `tox`).

**Comment by kloczek** [comment]
That is on what I'm trying to point because if one unit relieso on another unit it dissalows execute pytest by `pytest smome/file.py::some_test_unit`
Test suite should not have any dependencies between units and there are even tools to catch all that kind of dependencies such as https://github.com/mrbean-bremen/pytest-find-dependencies/

Unix tox is not viable solution because it executes test suite always in ixed env of other modules.
I want to have in my build procedures flexibility to execute pytest in altered set of modules to perform additional test builds for example to test all build modules with coverity tests. I can do that by insatall `pytest-cov` module and alter %pytest macto to use `--cov` switch. There are plenty of other pytest extensions which which could be used in such scenarios to asses quality of all modules on massive scalle.
All that woud be not possible if I'll be using tox instead pytest.
```console
[tkloczko@devel-g2v SPECS]$ ls -1 python-*.spec|wc -l; grep ^%pytest python-*.spec|wc -l
994
969
```
In other words I've manage to use pytest with almost 98% of all my packaged python modules.

**Comment by nicoddemus** [comment]
I'm not suggesting to use `tox`, I merely pointed out that you can look at the `tox.ini` file to see what dependencies the package officially runs its own test suite with.

**Comment by kloczek** [comment]
That is fine. I'm building all my packages in build envs spawned only to build sigle package in which are installed oinly build eependencies.
Still .. have in test suite dependencies between units is not good looking thing 🤔

**Comment by nicoddemus** [comment]
> Still .. have in test suite dependencies between units is not good looking thing 

I agree, but that's not what is happening here though.
